src/modules/onboard/onboard.pg.repository.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { DataSource, Repository } from 'typeorm';
+import { OnboardCategory } from '@src/infrastructure/database/entities/onboard-category.entity';
+import { onBoardCategoryList } from './onboard.const';
+
+@Injectable()
+export class OnBoardRepository extends Repository<OnboardCategory> {
+  constructor(private dataSource: DataSource) {
+    super(OnboardCategory, dataSource.createEntityManager());
+  }
+
+  private async addOnboardList() {
+    const newList = onBoardCategoryList.map((category) => ({ category }));
+
+    await this.createQueryBuilder()
+      .insert()
+      .into('onboards')
+      .values(newList)
+      .execute();
+
+    return this.find();
+  }
+
+  async getOnboardCategoryList() {
+    const categories = await this.find();
+
+    if (categories.length === 0) {
+      return this.addOnboardList();
+    }
+    return categories;
```

Comment:
아님 추가하는 로직을 넣어주고 싶다면 get에서 조회하는게 좋아보여
실제 조회를 추가하는 리턴값을 추가 add 메서드를 통해서 받고있는거 같아서!
나중에 추가메서드가 필요없다고 생각하면 지워도 조회에는 문제가 없으니까 ㅎ_ㅎ
```suggestion
  private async addOnboardList():promise<OnboardCategory[]>  {
    const newList = onBoardCategoryList.map((category) => ({ category }));

    await this.createQueryBuilder()
      .insert()
      .into('onboards')
      .values(newList)
      .execute();
  }

  async getOnboardCategoryList():Promise<OnboardCategory[]> {
    const categories = await this.find();

    if (categories.length === 0) {
       this.addOnboardList();
    }
     return this.find();
```

Reply:
근데 위에 같이 하면 반드시 (리스트 요소가 0개가 아니더라두)
find를 2번 부르는 것 같아서 좋아보이진 않넹

형 말대로 관심사 분리는 잘된 것 같은데, 2번 불리는게 신경이 쓰이네용

Reply:
아냐 석주 너말처럼 무조건 2번 조회하는건 좋은게 아니지!!

어차피 온보딩에서 가져올 데이터가 많지않아서 
저렇게 suggestion 남긴거고
만약 가져오는 데이터가 많다면 완전 최악☹️

최적화한다면 if 조건문외에 else로 최초에 받아온값을 리턴하면 될 것 같아!! (다만 코드가 좀 지저분해지네🥲)